webui/react/src/components/kit/ClipboardButton.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+import Icon from 'components/kit/Icon';
+import { copyToClipboard } from 'shared/utils/dom';
+import { notification } from 'utils/dialogApi';
+
+import Button from './Button';
+import Tooltip, { Placement } from './Tooltip';
+
+type IconName = 'clipboard' | 'checkmark';
+
+interface Props {
+  copiedMessage?: string;
+  disabled?: boolean;
+  // A get content function is used to dynamically fetch content when it is needed.
+  getContent: () => string;
+  tooltipPlacement?: Placement;
+  onCopy?: () => void;
+}
+
+export const TOOLTIP_LABEL_DEFAULT = 'Copy to clipboard';
+export const TOOLTIP_LABEL_SUCCESS = 'Copied!';
```

Comment:
**nit:** does this constant need to be exported?

Reply:
nope, I'll remove the `export`. I think I was using it for a test at one point but now it's not being used there. good catch